core/src/main/java/org/apache/spark/util/ChildFirstURLClassLoader.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.apache.spark.util;
+
+import sun.misc.CompoundEnumeration;
```

Comment:
I don't think we want to use this sun.misc class. Is there another way to stitch them together?

Reply:
There is `jodd.util.collection.CompositeEnumeration` but for that `jodd-core` must be added as a new dependency for spark-core. It is already in `dependencyManagement` so we are already using it Spark (to be precise in spark-hive). 

With that one even the `SuppressWarnings ` can be avoided too.

Reply:
Just build up an old `java.util.Vector` from the elements of the enumeration and get an enumeration from it? no need for a library.

Reply:
My primary reason of using ``CompoundEnumeration`` was to keep it consistent with ``java.lang.ClassLoader#getResources`` so that returned object is still an instance of ``sun.misc.CompoundEnumeration``

I do not oppose changing it but just for curiosity, any specific reason that we need to avoid ``CompoundEnumeration``.?? (one i can think of is ``sun.misc.CompoundEnumeration`` being dropped from java 9 and moved as nested class in ``java.lang.ClassLoader``)

Reply:
Yeah, that's it. I just don't want to depend on sun.misc, especially if you've found it doesn't exist in Java 9+. It's a good point but unless we know it won't work otherwise, I think we don't need the internal impl.

Reply:
Found a cleaner way using ``com.google.common.collect.Iterators`` i.e guava. Please review

Reply:
That's .. probably OK, though I remain twitchy about using Guava, given that it's something we have to shade and it becomes a dependency the classloader itself needs. I don't know if that has weird interactions. My instinct is to keep this as simple and free of possible conflicts as possible

Reply:
Agree with your opinion. So i now moved to using ``java.util.Collections`` only. Does it seem ok now.?